static/js/apps/eval_rig/eval_section.tsx

```diff
@@ -14,121 +14,143 @@
+  };
+
+  // Button Actions
+  const prevQuery = async () => {
+    if (await checkAndSubmit()) {
+      let targetId = sessionQueryId - 1;
+      while (!(targetId in allQuery)) {
+        targetId -= 1;
+      }
+      setSessionQueryId(targetId);
+      setSessionCallId(1);
     }
-    const sheet = doc.sheetsByTitle[DC_CALL_SHEET];
-    const rowIdx = props.calls[callPos].row;
-    sheet.getRows({ offset: rowIdx - 1, limit: 1 }).then((rows) => {
-      const row = rows[0];
-      if (row) {
-        setEvalInfo({
-          question: row.get(DC_QUESTION_COL),
-          dcResponse: row.get(DC_RESPONSE_COL),
-          llmStat: row.get(LLM_STAT_COL),
-          dcStat: row.get(DC_STAT_COL),
-        });
+  };
+  const prev = async () => {
+    if (await checkAndSubmit()) {
+      if (sessionCallId > 1) {
+        setSessionCallId(sessionCallId - 1);
       }
-    });
-  }, [doc, props.calls, callPos]);
-
-  // Highlight the current DC Call in the answer section.
-  useEffect(() => {
-    // Remove highlight from previous annotation
-    if (prevHighlightedRef.current) {
-      prevHighlightedRef.current.classList.remove("highlight");
     }
-
-    // Highlight the new annotation. Note the display index is 1 based.
-    const newHighlighted = document.querySelector(
-      `.annotation-${callPos + 1}`
-    ) as HTMLSpanElement;
-    if (newHighlighted) {
-      newHighlighted.classList.add("highlight");
-      prevHighlightedRef.current = newHighlighted;
+  };
+  const next = async () => {
+    if (await checkAndSubmit()) {
+      if (sessionCallId < numCalls()) {
+        setSessionCallId(sessionCallId + 1);
+      }
     }
-  }, [callPos]);
-
-  const previous = () => {
-    if (callPos > 0) {
-      setCallPos(callPos - 1);
+  };
+  const nextQuery = async () => {
+    if (await checkAndSubmit()) {
+      let targetId = sessionQueryId + 1;
+      while (!(targetId in allQuery)) {
+        targetId += 1;
+      }
+      setSessionQueryId(targetId);
+      setSessionCallId(1);
     }
   };
 
-  const next = () => {
-    if (callPos < props.calls.length - 1) {
-      setCallPos(callPos + 1);
-    }
+  // Button Conditions
+  const showPrevQuery = (): boolean => {
+    return sessionCallId == 1 && sessionQueryId > 1;
+  };
+  const showPrev = (): boolean => {
+    return sessionCallId > 1;
+  };
+  const showNext = (): boolean => {
+    return sessionCallId < numCalls();
+  };
+  const showNextQuery = (): boolean => {
+    return (
+      // When a query does not have any calls, the call id is 1 and num calls
+      // is 0.
+      sessionCallId >= numCalls() &&
+      sessionQueryId < Object.keys(allQuery).length
+    );
   };
 
   return (
     <>
-      {evalInfo && props.calls[callPos] && (
-        <FeedbackForm
-          queryId={props.queryId}
-          callId={props.calls[callPos].id}
-          evalInfo={evalInfo}
-        />
+      <FeedbackForm ref={formRef} />
+      {formRef.current && (
```

Comment:
nit: I feel like the ref passing is slightly messy and unnecessary, would it be easier/cleaner to just move the buttons to feedback form? OR make the navigation buttons their own component & make it a child of feedback form?

Reply:
You are right. I merge the two components together now. Can split if it gets too big later.